pkg/sqlcache/informer/listoption_indexer.go

```diff
@@ -200,7 +202,12 @@ func NewListOptionIndexer(ctx context.Context, s Store, opts ListOptionIndexerOp
 	l.RegisterBeforeDropAll(l.dropFields)
 	columnDefs := make([]string, len(indexedFields))
 	for index, field := range indexedFields {
-		column := fmt.Sprintf(`"%s" TEXT`, field)
+		typeName := "TEXT"
```

Comment:
Can we add some tests in listoption_indexer_test.go? Sounds like a `TestTypeGuidance` test would be useful here to see that part work (eg: perhaps with sorting as well, like sorting/filtering on a number?) Does that make sense?

Reply:
I just added one test with a non-empty TypeGuidance field to verify that the field type is "INT" and not "TEXT". But I don't think we have a unit-test framework that can do a full test that verifies sorting/filtering on a number behaves as expected. If I'm reading the code right, `listoption_indexer_test.go` isn't aware of which kind of database table it's working with; it just passes the fields. So there's no processing `TypeGuidance` there. And we don't have any other unit tests that call `loi.ListByOptions()`.

Any thoughts?